### DISCUSSION TITLE
Authentication enabled but still able to connect to test collection without username and password

### DISCUSSION THREAD

**Discussion by akash-swami-invimatic-io** [docker-library/mongo#632] [General]
I am running mongo docker using docker compose file with below configurations.

```shell
version: '3.1'
services:
  mongo:
    image: mongo
    restart: always
    environment:
      MONGO_INITDB_ROOT_USERNAME: root
      MONGO_INITDB_ROOT_PASSWORD: some_password
    ports:
      - "127.0.0.1:27017:27017"
```

when I exec to mongo container and connect to mongodb through mongosh command without mentioning username and password its still connecting. Based on my understanding when authentication is enabled it should not connect to test database as well.

<img width="583" alt="image" src="https://github.com/docker-library/mongo/assets/127176287/09db8f3d-0ecb-444f-a2ed-d9d0faa58bcf">

I am just started working with mongodb so please let me know if this is expected behavior or I am missing some configuration while creating container.

**Comment by tianon** [comment]
This is how MongoDB works, AFAIK - it has to allow you to connect so that you can then authenticate (against the correct database too, since that's a factor in the authentication), and the authentication is a normal command against the server.  You shouldn't be able to do anything "interesting" with the server you've connected to until you've authenticated, though.
